File: node_sample/payload/SendingUrl.js
var unirest = require('unirest');
const Recipients = require('./Recipients');
const CC = require('./CC');

class SendingUrl {
    recipients = []
    cc = []

	constructor(file_id, recipients, message='', redirect_url='', cc=null, type='')
	{
		this.file_id = file_id;
		this.add_recipients(recipients);
        this.message = message;

        if (cc != null)
        {
            this.add_cc(cc)
        }
        this.type = type;
        this.redirect_url = redirect_url;

      }

      add_recipients(recipients)
      {
        if (!(recipients instanceof Recipients))
        {
            throw new Error("Invalid type of recipients");
        }
        this.recipients.push(recipients);
      }

      add_cc(cc)
      {
        if (!(cc instanceof CC))
        {
            throw new Error("Invalid type of cc");
        }
        this.cc.push(cc);
      }

      to_json()
      {
            return  {"file_id": this.file_id, "type": this.type, "message": this.message,
                     "recipients": this.recipients, "cc": this.cc, "redirect_url": this.redirect_url}
      }

      fetch_embedded_sending_url = function (base_url, api_token) {
        var payload = this.to_json();
        var url = base_url + "v2/signing/url/";
        return new Promise((resolve, reject) => {
        unirest.post(url)
        .headers({'Authorization': api_token, 'Content-Type': 'application/json'})
        .send(payload)
        .end(function (response) {
        if (response.error) {
        return reject(response.body)
        }
        return resolve(response.body);
        });
    })
    }
}

module.exports = SendingUrl

File: java_samples/payloads/SendingUrl.java
package payloads;

import java.util.ArrayList;

import org.apache.http.client.methods.CloseableHttpResponse;
import org.apache.http.client.methods.HttpPost;
import org.apache.http.entity.StringEntity;
import org.apache.http.impl.client.CloseableHttpClient;
import org.apache.http.impl.client.HttpClients;
import org.json.JSONObject;

public class SendingUrl {
	private String type = "";
	private ArrayList<Recipients> recipients = new ArrayList<Recipients>();
	private String message = "";
	private ArrayList<CC> cc = new ArrayList<CC>();
	private String file_id;
	private String redirect_url = "";

	public SendingUrl(String file_id, Recipients recip)
	{
		this.file_id = file_id;
		this.recipients.add(recip);
	}
	
	public void set_message(String msg) 
	{
		this.message = msg;
	}
	
	public void add_cc(CC obj) 
	{
		this.cc.add(obj);
	}
	
	public void set_type(String msg) 
	{
		this.type = msg;
	}
	
	public void set_redirect_url(String msg) 
	{
		this.redirect_url = msg;
	}
	
	public void add_recipients(Recipients obj) 
	{
		this.recipients.add(obj);
	}
	
	public JSONObject toJson() 
	{
		JSONObject obj = new JSONObject();

		ArrayList<JSONObject> array = new ArrayList<JSONObject>();
		
		for (int i = 0; i < this.recipients.size(); i++)  
		{
            array.add(this.recipients.get(i).toJson());         
		} 
		obj.put("recipients", array);
		array.clear();
		
		for (int i = 0; i < this.cc.size(); i++)  
		{
            array.add(this.cc.get(i).toJson());         
		} 
		obj.put("cc", array);
		array.clear();
		
		
		obj.put("file_id", this.file_id);
		obj.put("type", this.type);
		obj.put("message", this.message);
		
		
		obj.put("redirect_url", this.redirect_url);
		
		return obj;
		
	}
	
	public CloseableHttpResponse fetch_embedded_sending_url(String base_url, String api_token) throws Exception {

        HttpPost post = new HttpPost(base_url + "v2/signing/url/");
        post.addHeader("content-type", "application/json");
        post.addHeader("Authorization", api_token);
        post.setEntity(new StringEntity(this.toJson().toString()));

        CloseableHttpClient httpClient = HttpClients.createDefault();
        CloseableHttpResponse response = httpClient.execute(post);

        return response;
    }
	
}
